src/main/java/net/sf/jabref/gui/PreviewPanel.java

```diff
@@ -275,16 +275,27 @@ public void setLayout(Layout layout) {
     }
 
     public void setEntry(BibEntry newEntry) {
-        if(entry.isPresent() && (entry.get() != newEntry)) {
-            entry.ifPresent(e -> e.removePropertyChangeListener(this));
-            newEntry.addPropertyChangeListener(this);
+
+        if (entry.isPresent() && (entry.get() != newEntry)) {
+            entry.ifPresent(e -> e.unregisterListener(this.currentChangeFieldUpdateEvent));
+            this.currentChangeFieldUpdateEvent = new ChangeFieldUpdateEvent();
+            newEntry.registerListener(this.currentChangeFieldUpdateEvent);
         }
+
         entry = Optional.ofNullable(newEntry);
 
         updateLayout();
         update();
     }
 
+    private class ChangeFieldUpdateEvent {
```

Comment:
Yes it is necessary, cause there is only one `PreviewPanel` for all entries. There should be a possibility to replace the listener and to set new listener for new entries. Furthermore passing small objects (like `ChangeFieldUpdateEvent`) is more efficient than passing the hole `PreviewPanel` object.